### DISCUSSION TITLE
Cobra CLI in Go: How to attach additional flags to the "serve" command?

### DISCUSSION THREAD

**Discussion by ramadhanafif** [pocketbase/pocketbase#4304] [Q&A] [answered]
Hi, I have added certain functionality to the pocketbase, and I want to enable it through cobra CLI. As the functionality is tied to the DB, I want to attach it as an optional boolean flag, in the serve command. Therefore I can do this:

```sh
./pocketbase serve --http 1.1.1.1:1234 --feature --additionalFlag flagParams
```

**Comment by ganigeorgiev** [comment, marked as answer]
You can explore the prebuilt executable [example/base/main.go file](https://github.com/pocketbase/pocketbase/blob/master/examples/base/main.go) to see how global flags are registered.

If you want to apply the flags only for the `serve` command you'll have to iterate over `app.RootCmd.Commands()` (_edit: this actually may not work as the `serve` command is registered as part of the `Start()` method but you can always check manually `os.Args` and load the flags conditionally_).

**Reply by ramadhanafif** [reply]
Yes, I have tried to iterate over the `Commands()` but it returns nothing, since the `serve` has not been added to the commands list.

It seems that if I want to do this, I must insert the custom flags after `NewServeCommand` and prior to `execute()` commands. But it is too much of a hassle, and might as well add it as a root flag.

Thanks for the response, and thanks for responding fast.
